pandas/tools/plotting.py

```diff
@@ -816,7 +816,12 @@ def __init__(self, data, kind=None, by=None, subplots=False, sharex=True,
             grid = False if secondary_y else True
 
         self.grid = grid
-        self.legend = legend
+        if legend is not None:
```

Comment:
The original point of this was decide if a label should be rendered based on whether there was a valid label provided or if there was more than 1 series. I can revert back the defaults, though, and make legend show default series names of no legend is provided.